atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -37,41 +37,124 @@
+     * applies to all of the rows.
+     *
+     * If the provided {@link Iterable} of {@code rows} has a stable ordering, the returned iterator is guaranteed
+     * to return cell-value pairs in a lexicographic ordering over rows and columns, where rows are sorted according to
+     * the stable ordering of {@code rows}, and columns are sorted on byte ordering. If {@code rows} does not have a
+     * stable ordering, the returned iterator will return cell-value pairs that are ordered within a row - that is,
+     * columns from the same row are always grouped together and are always sorted on byte ordering, but there are no
+     * guarantees on the ordering of rows.
+     *
+     * @param tableRef table to load values from
+     * @param rows unique rows to apply the column range selection to
+     * @param columnRangeSelection range of columns to load for each of the rows provided
+     * @param batchHint number of columns that should be loaded from the underlying database at once
+     * @return an iterator over cell-value pairs, guaranteed to follow the ordering outlined above
+     * @throws IllegalArgumentException if {@code rows} contains duplicates
+     */
     @Idempotent
     Iterator<Map.Entry<Cell, byte[]>> getRowsColumnRange(
             TableReference tableRef,
             Iterable<byte[]> rows,
             ColumnRangeSelection columnRangeSelection,
             int batchHint);
 
+    /**
+     * Returns a mapping of rows to {@link Iterator}s over cell-value pairs within {@code tableRef} for the specified
+     * {@code rows}, where the columns fall within the provided {@link BatchColumnRangeSelection}. The single provided
+     * {@link BatchColumnRangeSelection} applies to all of the rows.
+     *
+     * The returned {@link BatchingVisitable}s are guaranteed to return cells matching the predicate. These are sorted
+     * by column, with ascending byte ordering.
+     *
+     * It is guaranteed that the {@link Map#keySet()} of the returned map has a corresponding element for each of the
+     * input {@code rows}, even if there are rows where no columns match the predicate.
+     *
+     * @param tableRef table to load values from
+     * @param rows unique rows to apply the column range selection to
+     * @param columnRangeSelection range of columns and batch size to load for each of the rows provided
+     * @return a mapping of rows to cells matching the predicate in the row, following the ordering outlined above
+     * @throws IllegalArgumentException if {@code rows} contains duplicates
+     */
     @Idempotent
     Map<byte[], Iterator<Map.Entry<Cell, byte[]>>> getRowsColumnRangeIterator(
             TableReference tableRef,
             Iterable<byte[]> rows,
             BatchColumnRangeSelection columnRangeSelection);
 
+    /**
+     * Gets the values associated for each cell in {@code cells} from table specified by {@code tableRef}.
+     *
+     * @param tableRef the table from which to get the values
```

Comment:
Should this also have something around Cell lookup from this map? I guess that's implied because Cell has an #equals we control.

Reply:
I could add it, though yeah I didn't because Cell#equals() behaves predictably